operator/pkg/upgrade/upgrade.go

```diff
@@ -202,6 +202,26 @@ func updateClusterConfig(ctx context.Context, cli client.Client, in *ecv1beta1.I
 		}
 	}
 
+	// Apply unsupported overrides from the installation
+	if (in.Spec.Config != nil && in.Spec.Config.UnsupportedOverrides.K0s != "") || in.Spec.EndUserK0sConfigOverrides != "" {
+
+		newCfg, err := config.PatchK0sConfig(&currentCfg, in.Spec.Config.UnsupportedOverrides.K0s, true)
```

Comment:
wouldn't this panic if `in.Spec.Config` is nil? maybe have two separate if blocks?

Reply:
ah whoops